gateways/clients/responses/hathor_wallet_service/hathor_wallet_service_token_response.py

```diff
@@ -0,0 +1,161 @@
+    timestamp: int
+
+
+@dataclass
+class HathorWalletServiceTokenTransactionsReponse:
+    """Response from hathor wallet service when requesting a list of token transactions
+
+    :param total: total number of transactions of a given token
+    :type total: int
+    :param transactions: list of transactions
+    :type transactions: List[:py:class:`domain.tx.TokenTransactionListItem`]
+    """
+    total: int
+    transactions: List[HathorWalletServiceTokenTransaction]
+
+    @classmethod
+    def from_dict(cls, dikt: dict) -> 'HathorWalletServiceTokenTransactionsReponse':
+        return from_dict(data_class=cls, data=dikt)
+
+    def to_token_transaction_list(self) -> TokenTransactionList:
+        """Convert wallet service response to a domain token transaction list
+
+        :raises Exception: If the token transacton list is in wrong format
+        :return: the new TokenTransactionList generated
+        :rtype: :py:class:`domain.tx.token.TokenTransactionList`
+        """
+        try:
+            return TokenTransactionList(
+                total=self.total,
+                transactions=[TokenTransactionListItem(
+                    id=transaction.id,
+                    timestamp=transaction.timestamp
+                ) for transaction in self.transactions]
+            )
+        except Exception:
+            raise Exception('malformed_token_transaction_list')
+
+
+@dataclass
+class HathorWalletServiceToken:
+    """Item of a token list with token info
+
+    :param id: id hash of the token
+    :type id: str
+    :param name: token name
+    :type name: str
+    :param symbol: token symbol
+    :type symbol: str
+    :param timestamp: timestamp of token creation
+    :type timestamp: int
+
+    """
+    id: str
+    name: str
+    symbol: str
+    timestamp: int
+
+
+@dataclass
+class HathorWalletServiceTokenListResponse:
+    """Response from hathor wallet service when requesting a list of tokens
+
+    :param total: total number of tokens
+    :type total: int
+
+    :param tokens: list of tokens
+    :type tokens: List[:py:class:`domain.tx.TokenListItem`]
+    """
+    total: int
+    tokens: List[HathorWalletServiceToken]
+
+    @classmethod
+    def from_dict(cls, dikt: dict) -> 'HathorWalletServiceTokenListResponse':
+        return from_dict(data_class=cls, data=dikt)
+
+    def to_token_list(self) -> TokenList:
+        """Convert wallet service response to a domain token list
+
+        :raises Exception: If the token list is in wrong format
+        :return: the TokenList generated
+        :rtype: :py:class:`domain.tx.token.TokenList`
+        """
+        try:
+            return TokenList(
+                total=self.total,
+                tokens=[TokenListItem(
+                    id=token.id,
+                    name=token.name,
+                    symbol=token.symbol,
+                    timestamp=token.timestamp
+                ) for token in self.tokens]
+            )
+        except Exception:
```

Comment:
It feel dangerous to catch `Exception`. Why is that? It would capture ANY exception generated inside it (even if your code is broken).